### PR TITLE
Append the missing function name"GetAppletInfo", "SetAppCpuTimeLimit" and "GetAppCpuTimeLimit" to APT:A

### PR DESCRIPTION
**Pull request by JamePeng**
The APT::GetAppletInfo was implemented in #1266, just missing the function name in APT:A
qq 20160327172305

### CODE REVIEW

**Comment by wwylele** [comment]
Ah, were to PR this then found one already here.
Also, add `GetAppletManInfo` and some other functions?

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -397,6 +397,25 @@ void GetAppletInfo(Service::Interface* self) {
...
+    StartupArgumentType startup_argument_type = static_cast<StartupArgumentType>(cmd_buff[2]);
+
+    if (parameter_size >= 0x300) {
+        LOG_ERROR(Service_APT, "Parameter was out of range!(capped to 0x300) parameter_size=0x%08x", parameter_size);
+        return;
+    }
+
+    u32* parameter = (u32*)Memory::GetPointer(cmd_buff[0x41]);
+
+    LOG_WARNING(Service_APT,"startup_argument_type=%u , parameter_size=0x%08x , parameter_value=0x%08x",
```
Comment (wwylele) [inline src/core/hle/service/apt/apt.cpp]: add `"(stubbed) called"`

**Comment by JamePeng** [comment]
Ready for merge!

File: src/core/hle/service/apt/apt.cpp [hunk 2]
```diff
@@ -397,6 +397,25 @@ void GetAppletInfo(Service::Interface* self) {
...
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 parameter_size = cmd_buff[1];
+    StartupArgumentType startup_argument_type = static_cast<StartupArgumentType>(cmd_buff[2]);
+
+    if (parameter_size >= 0x300) {
+        LOG_ERROR(Service_APT, "Parameter was out of range!(capped to 0x300) parameter_size=0x%08x", parameter_size);
+        return;
+    }
+
+    u32* parameter = (u32*)Memory::GetPointer(cmd_buff[0x41]);
```
Comment (bunnei) [inline src/core/hle/service/apt/apt.cpp]: What is the purpose of this? If you want to log whats at this address, use Memory::Read32, not GetPointer

File: src/core/hle/service/apt/apt.cpp [hunk 3]
```diff
@@ -397,6 +397,25 @@ void GetAppletInfo(Service::Interface* self) {
     LOG_WARNING(Service_APT, "(stubbed) called appid=%u", app_id);
 }
 
+void GetStartupArgument(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    u32 parameter_size = cmd_buff[1];
+    StartupArgumentType startup_argument_type = static_cast<StartupArgumentType>(cmd_buff[2]);
+
+    if (parameter_size >= 0x300) {
+        LOG_ERROR(Service_APT, "Parameter was out of range!(capped to 0x300) parameter_size=0x%08x", parameter_size);
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.cpp]:
```cpp
"Parameter is out of range (capped to 0x300): parameter_size=0x%08x"
```

File: src/core/hle/service/apt/apt.h [hunk 4]
```diff
@@ -67,6 +67,12 @@ enum class AppletId : u32 {
     Ed2                = 0x402,
 };
 
+enum StartupArgumentType {
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.h]:
```cpp
enum class StartupArgumentType {
    OtherApp   = 0,
    Restart    = 1,
    OtherMedia = 2,
};

**Comment by JamePeng** [comment]
The function SetAppCpuTimeLimit and GetAppCpuTimeLimit was implemented too!
qq 20160405022034

**Comment by lioncash** [comment]
@JamePeng OK, we've been over this before with regards to you just adding things to a pull request that should be in a different pull request or completely don't correspond with what the title indicates – we've been over this four times now (including this). four times – a nice and solid 4.

Can you please stop doing it? I genuinely don't mind the first few times. Everyone's done it at some point. But this is literally a *consistent* thing you do, and it makes a few other devs not want to review changes you submit.

If your PR title says `Append the missing function name"GetAppletInfo" to APT:A` the only thing anyone wanting to review your PR expects is GetAppletInfo being added to the APT:A table.

**Comment by JamePeng** [comment]
so I have to reset them?

File: src/core/hle/service/apt/apt.h [hunk 5]
```diff
@@ -344,6 +350,17 @@ void PreloadLibraryApplet(Service::Interface* self);
  */
 void StartLibraryApplet(Service::Interface* self);
 
+/**
+* APT::GetStartupArgument service function
```
Comment (bunnei) [inline src/core/hle/service/apt/apt.h]: Indentation is wrong

**Comment by JamePeng** [comment]
done!

**Comment by bunnei** [comment]
LGTM

File: src/core/hle/service/apt/apt.cpp [hunk 6]
```diff
@@ -397,6 +397,23 @@ void GetAppletInfo(Service::Interface* self) {
...
+    u32 parameter_size = cmd_buff[1];
+    StartupArgumentType startup_argument_type = static_cast<StartupArgumentType>(cmd_buff[2]);
+
+    if (parameter_size >= 0x300) {
+        LOG_ERROR(Service_APT, "Parameter size is outside the valid range (capped to 0x300): parameter_size=0x%08x", parameter_size);
+        return;
+    }
+
+    LOG_WARNING(Service_APT,"(stubbed) called startup_argument_type=%u , parameter_size=0x%08x , parameter_value=0x%08x",
+                startup_argument_type, parameter_size, Memory::Read32(cmd_buff[41]));
```
Comment (mailwl) [inline src/core/hle/service/apt/apt.cpp]: 41 - must be in hex (0x41). And it's OUTPUT buffer, no reason to read from it

**Pull request merged**